tests/lib/rules/no-sync.js

```diff
@@ -149,3 +149,88 @@ new RuleTester().run("no-sync", rule, {
         },
     ],
 })
+
+new TsRuleTester().run("no-sync", rule, {
```

Comment:
Could you add some tests for the `from: package` and `from: lib` ignores too please?

Reply:
I added one for lib. I'm not sure if we can easily add one for package though

Reply:
More tests added.